### DISCUSSION TITLE
Electrolysis (gladiator)

### DISCUSSION THREAD

**Discussion by V-ktor** [redeclipse/base#1134] [Maps and Content Showcase]
This is a gladiator map featuring 54 platforms that are dropping randomly into deadly water.
Bots have big problems as they don't understand that they are not allowed to step on deactivated platforms.
[electrolysis.zip](https://github.com/red-eclipse/world/files/1684939/electrolysis.zip)

![20180131203717](https://user-images.githubusercontent.com/12966454/35643766-8bcab708-06c7-11e8-87c3-ffc4ada96e15.jpg)
![20180131203636](https://user-images.githubusercontent.com/12966454/35643768-8e31bc12-06c7-11e8-8192-09b0a1bdd294.jpg)

**Comment by robalni** [comment]
That's a pretty cool map! Looks good and a fun idea.

Some things I noticed:

The uneven floor makes it hard to jump sometimes. (if you try to jump as you step on a pole)

The water and platforms (and everything else) have almost the same color. It makes it a bit hard to see.

The water is very close to the platforms so you don't really notice that you fall off before you're dead. What about lowering the water?

It would be nice if it was possible to see that the platform is about to go down before it does so you don't get a surprise.

**Comment by SniperGoth** [comment]
Looks like someone had the same idea as mine... but actually made a map for it, hahah.
My personal suggestions would be to make it more colorful, and as robalni said, lowering the water a bit.
I like the script you made where things trigger randomly... it's gonna be useful ♐️ .

**Comment by V-ktor** [comment]
I lowered the water a bit and added red light cones one second before the platforms go down.
Also the lights are now orange.
[electrolysis.zip](https://github.com/red-eclipse/world/files/1683429/electrolysis.zip)
![20180131224244](https://user-images.githubusercontent.com/12966454/35649170-7d607cf0-06d8-11e8-8eb2-61ec6c333b2f.jpg)

**Comment by no-lex** [comment]
Many players (like me) who have volumetrics disabled are not going to see the light effects. If you would like to have the map included in 2.0, it would have to be modified to conform to the guidelines in https://github.com/red-eclipse/maps/issues/48. 

Also, the ambient lighting isn't properly flagged. Consider setting at least one of the two lights hanging from each ceiling corner to noshadow nospecular (10) as doing so will improve performance on such a water-heavy map while having a negligible effect on visuals.


If you fix those few issues, I wouldn't have a problem including this in 2.0; it's a nicely made gladiator map. It meets the <s>only requirement</s> main requirement for a good gladiator map: a unique feature that makes the map more than a floating platform.

**Comment by cloudziq** [comment]
Good map, but there should be a second to react, it is NOT fun to die instantly, it's now a too much luck based in my opinion
If I were you, I'll make these platforms a little different, with lights showing for only second before platform moves), also I suggest to make these platforms to travel more distance, like 128 or 64 grid units, so these can be reused in other new maps as important element of gameplay :)

**Comment by V-ktor** [comment]
I set the flags of the 4 upper lights to 10 and added particles that appear before the platforms go down.
Also the death material was replaced by hurt material (56 damage) with stun.
[electrolysis.zip](https://github.com/red-eclipse/world/files/1684939/electrolysis.zip)
![20180201103403](https://user-images.githubusercontent.com/12966454/35671795-cf8246f8-073c-11e8-8c90-f6e3148b2989.jpg)

There are two versions of the platform animation (and another one was used in my other map redeclipse/world#34). I could add more animations with different distances and going up or down, but platforms don't work. They are just set to noclip like doors are.

**Comment by qreeves** [comment]
I love the concept, might actually make for some interesting gladiator matches.
